Applications/Custom/LayerClient/jni/pow.cpp

```diff
@@ -89,7 +89,7 @@ int PowLayer::setProperty(std::vector<std::string> values) {
   return nntrainer::Layer::setProperty(unhandled_values);
 }
 
-int PowLayer::initialize(nntrainer::Manager &manager) {
+int PowLayer::initialize(std::shared_ptr<nntrainer::Manager> manager) {
```

Comment:
I wonder where the copy constructor is called. 
In my understanding, passing the reference does not invoke copy constructor.
Also, ownership is with the neuralnetwork with `Manager manager`, and passed reference will be invalidated after model class is done with. But with `shared_ptr<Manager>`, layer can create a copy and keep it alive.
I dont understand why have a shared_ptr here (and it overhead), can you please elaborate on this.

Reply:
We made an offline discussion to check if this can be reverted.

Reply:
Reverted